packages/lit-html/src/directives/ref.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {directive, nothing, ElementPart} from '../lit-html.js';
+import {DisconnectableDirective} from '../disconnectable-directive.js';
+
+/**
+ * Creates a new Ref object, which is container for a reference to an element.
+ */
+export const createRef = <T = Element>() => new Ref<T>();
+
+class Ref<T = Element> {
+  /**
+   * The current Element value of the ref, or else `undefined` if the ref is no
+   * longer rendered.
+   */
+  value?: T;
+
+  /**
+   * Updates the element value of this Ref. Users should generally not need to
+   * call this function; it will be called automatically when passed to the
+   * `ref()` directive.
+   * @param element
+   */
+  set(element: T | undefined): void {
```

Comment:
Not sure why we have this if we're not doing `fowardingRef`.

Reply:
My thought was we could eventually make Ref observable to trigger re-rendering when you forwarded it. We can do that later though, and with a setter.

Reply:
Removed for now and made `readonly`.